src/utils/__tests__/symlink.test.ts

```diff
@@ -0,0 +1,235 @@
+import { createSymlinks } from '../symlink';
+import { withTempDir } from '../files';
+import { promises as fsPromises } from 'fs';
+
+import * as path from 'path';
+import * as fs from 'fs';
+
+describe('createSymlinks', () => {
+  it.each([
+    {
+      _name: 'major version upgrade',
+      oldVersion: '0.9.0',
+      newVersion: '1.0.0',
+      expectedSymlinks: ['1.json', '1.0.json', 'latest.json'],
+    },
+    {
+      _name: 'minor version upgrade',
+      oldVersion: '0.2.3',
+      newVersion: '0.3.0',
+      expectedSymlinks: ['0.json', '0.3.json', 'latest.json'],
+    },
+    {
+      _name: 'patch version upgrade',
+      oldVersion: '0.2.3',
+      newVersion: '0.2.4',
+      expectedSymlinks: ['0.json', '0.2.json', 'latest.json'],
+    },
+    {
+      _name: 'old version is undefined',
+      oldVersion: undefined,
+      newVersion: '1.2.3',
+      expectedSymlinks: ['1.json', '1.2.json', 'latest.json'],
+    },
+    {
+      _name: 'updating older major (patch)',
+      oldVersion: '2.3.4',
+      newVersion: '1.2.3',
+      expectedSymlinks: ['1.json', '1.2.json'],
+    },
+    {
+      _name: 'updating older major (minor)',
+      oldVersion: '2.3.4',
+      newVersion: '1.3.0',
+      expectedSymlinks: ['1.json', '1.3.json'],
+    },
+  ])('$_name', async ({ oldVersion, newVersion, expectedSymlinks }) =>
+    withTempDir(async tmpDir => {
+      const versionFile = `${newVersion}.json`;
+      const versionFilePath = path.join(tmpDir, versionFile);
+
+      createSymlinks(versionFilePath, newVersion, oldVersion);
+
+      // Check that there are no other files in the directory
+      const filesInDir = await fsPromises.readdir(tmpDir);
+      expect(filesInDir.sort()).toStrictEqual(expectedSymlinks.sort());
+
+      for (const symlinkFile of filesInDir) {
+        const symlinkFilePath = path.join(tmpDir, symlinkFile);
+
+        // Make sure it's a symbolic link
+        const fileStat = await fsPromises.lstat(symlinkFilePath);
+        expect(fileStat.isSymbolicLink()).toBe(true);
+
+        // Make sure it points to the right file
+        const symlinkDestination = await fsPromises.readlink(symlinkFilePath);
+        expect(symlinkDestination).toBe(versionFile);
+      }
+    }, true)
+  );
+
+  describe('correctly handles already existing symlinks', () => {
+    it('when updating an old major version', async () =>
```

Comment:
the `describe` / `it`s read backwards here -- I think they're supposed to be the other way around so the code "reads"?